src/content/docs/recipes/testing/snapshots.md

```diff
@@ -0,0 +1,140 @@
+---
+title: Testing with insta snapshots
+sidebar:
+  order: 1
+---
+
+When building terminal user interfaces (TUIs) with Ratatui, it's crucial to ensure your application
+renders as expected across different scenarios. One powerful tool for achieving this is
+[`cargo-insta`]. It allows you to capture and compare the terminal output over time, making it ideal
+for testing visual output like TUIs.
+
+In this guide, we'll show you how to set up snapshot testing for Ratatui-based applications.
+
+:::note[Why Snapshot Testing?]
+
+Snapshot testing captures a rendered "snapshot" of your TUI at a given point and compares it to an
+expected snapshot stored in your test suite. If there's a mismatch, the test fails, highlighting
+changes in your TUI's rendering, whether intentional or accidental.
+
+:::
+
+## Setting Up Snapshot Testing
+
+Here's how to structure your Ratatui application to use insta snapshots effectively.
+
+### 1. Add Dependencies
+
+First, make sure to include [`insta`] crate in your `Cargo.toml`:
+
+```toml
+[dev-dependencies]
+insta = "1.40"
+```
```

Comment:
Versions get out of date, use cargo add instead of mentioning Cargo.toml, and add instructions to install cargo-insta.
```suggestion
First, make sure to install cargo-insta and the include [`insta`] crate in your `Cargo.toml`:

```shell
cargo install cargo-insta
cargo add insta
```
```